webhooks/webhooks.go

```diff
@@ -0,0 +1,186 @@
+package webhooks
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"go.sia.tech/core/types"
+	"go.uber.org/zap"
+	"lukechampine.com/frand"
+)
+
+const (
+	webhookTimeout   = 10 * time.Second
+	WebhookEventPing = "ping"
+)
+
+type (
+	WebHookRegisterRequest struct {
+		Event Event  `json:"event"`
+		URL   string `json:"url"`
+	}
+	WebHookRegisterResponse struct {
+		ID types.Hash256 `json:"id"`
+	}
+
+	WebHook struct {
+		ID    types.Hash256 `json:"id"`
+		Event Event         `json:"event"`
+		URL   string        `json:"url"`
+	}
+
+	registeredHook struct {
+		Event
+		url string
+	}
+
+	Event struct {
+		Module  string      `json:"module"`
+		ID      string      `json:"id"`
+		Payload interface{} `json:"payload,omitempty"`
+	}
+)
+
+type Webhooks struct {
+	mu     sync.Mutex
+	hooks  map[types.Hash256]*registeredHook
```

Comment:
shouldn't we persist webhooks? seems like something I would want to have persisted, or at least i'd be very annoyed to re-register my webhooks whenever I restart my autopilot

Reply:
I'll f/u with at least 2 more PRs. One to move the alerts to the bus as well and then one to persist hooks.